### DISCUSSION TITLE
Akkuschonung niedriger priorisiert als BattVollUm/100%

### DISCUSSION THREAD

**Discussion by iReckyy** [wiggal/GEN24_Ladesteuerung#109] [Q&A]
Die Akkuschonung wird ausgesetzt wenn BattVollUm oder 100% erreicht wird.

Ich habe aktuell noch kleinere Probleme mit dem Akku, bekomme ihn aber nicht bis zur Wunschzeit (Sonnenuntergang -4h) voll.
Danach aber schon.

Wenn er jetzt aber mit MaxLadung lädt, schießen mir manche Zellen so schnell hoch, dass das Ladeende zu früh eingeleitet wird.

Kann man das evtl. wählbar machen oder ändern?
.
.
.
.
***** BEGINN: 2024-10-28 15:43:02 *****

## HTTP-LADESTEUERUNG ##
>>>Parameteränderung durch WebUI-Settings: Ladesteuerung:['logging', 'laden', 'optimierung', 'notstrom']
aktuellePrognose: 2051
TagesPrognose - BattVollUm: 0 - 13
Grundlast_Summe für Tag: 0
aktuellePVProduktion/Watt: 466
aktuelleEinspeisung/Watt: 0
aktuelleBatteriePower/Watt: -43
GesamtverbrauchHaus/Watt: 423
Akku_Zustand in Prozent: 95.0 %
aktuelleBattKapazität/Watt: 1420
Batteriestatus in Prozent: 85.4 %
LadewertGrund: BattVollUm oder Akkustand 100% erreicht!
Bisheriger Ladewert/Watt: 4000
Neuer Ladewert/Watt: 4000

Änderungen kleiner Schreibgrenze!

In SQLite-Datei gespeichert!

DEBUG *************** Berechnung Abzugswert:

DEBUG *************** Ladewertmittel LOOP: 0
DEBUG ########### Pro_Akt_fun: 433.78 REST_Akt_Minute: 0.283
DEBUG 2024-10-28 14:00:00 Pro_Akt_fun: 433.78 Prognose_gesamt: 433.78

DEBUG *************** Ladewertmittel LOOP: 1
DEBUG 2024-10-28 15:00:00 Pro_Akt_fun: 2540 Prognose_gesamt: 2973.78

DEBUG *************** Ladewertmittel LOOP: 2
DEBUG ########### Pro_Akt_fun: 1128.75 Akt_Minute: 0.717
DEBUG 2024-10-28 16:00:00 Pro_Akt_fun: 1128.75 Prognose_gesamt: 4102.53

DEBUG 10/28/24 15:43 Aktuelle Prognose - Reservierung: 2051, Batteriekapazität: 1420, aktuellerLadewert: 4000

DEBUG <<<<<< Meldungen von Akkuschonung >>>>>>>
DEBUG Akkuschonung_Werte: {'80': '0.2', '85': '0.15', '90': '0.1', '95': '0.05', '99': '0.025'}
DEBUG AkkuschonungLadewert-alterLadewert: 2464
DEBUG BattStatusProz_Grenze: 85
DEBUG AkkuschonungLadewert: 1536
DEBUG aktuellerLadewert: 4000

DEBUG BattVollUm: 13Uhr
DEBUG WRSchreibGrenze_nachUnten: 500W
DEBUG WRSchreibGrenze_nachOben: 250W

DEBUG <<<<<<<< Eigenverbrauchs-Optimierung >>>>>>>>>>>>>
DEBUG ## Dauer_Nacht_Std: 0.28, Akku_Rest_Watt: 7221
DEBUG ## Eigen_Opt_genau: 25559, Eigen_Opt_Std_neu: 25500
DEBUG ## >>> Bei PrognoseMorgen < PrognoseGrenzeMorgen, keine Einspeisung während des Tages
DEBUG ## >>> PrognoseMorgen: 24.777, PrognoseGrenzeMorgen: 40
DEBUG ## >>> EigenverbOpt_steuern == 2, keine Einspeisung während des Tages
***** ENDE: 2024-10-28 15:43:02 *****

**Comment by wiggal** [comment]
> Die Akkuschonung wird ausgesetzt wenn BattVollUm oder 100% erreicht wird.
> 

UND wenn aktuelle Produktion (aktuellePVProduktion/Watt: 466)  > Akkuschonwert (AkkuschonungLadewert: 1536), was hier nicht der Fall ist. Heißt in deinem Beispiel, es wird sowieso nichts mehr in den Akku geladen.

> > > > aktuelleBatteriePower/Watt: -43


> Ich habe aktuell noch kleinere Probleme mit dem Akku, bekomme ihn aber nicht bis zur Wunschzeit (Sonnenuntergang -4h) voll. Danach aber schon.

Sollte die Produktion wieder steigen, würde er die Akkuschonung einschalten.
 
> Wenn er jetzt aber mit MaxLadung lädt, schießen mir manche Zellen so schnell hoch, dass das Ladeende zu früh eingeleitet wird.
> 
>>>> DEBUG Akkuschonung_Werte: {'80': '0.2', '85': '0.15', '90': '0.1', '95': '0.05', '99': '0.025'} 

Ich würde dann evtl. etwas früher weniger laden (z.B. {'75': '0.1', ....)

**Comment by iReckyy** [comment]
> UND wenn aktuelle Produktion (aktuellePVProduktion/Watt: 466) > Akkuschonwert (AkkuschonungLadewert: 1536), was hier nicht der Fall ist. Heißt in deinem Beispiel, es wird sowieso nichts mehr in den Akku geladen.

Ok, dass hatte ich nicht auf dem Schirm, habe wohl zu einem blöden Zeitpunkt geguckt. Da waren halt andauernd Wolken/Sonne abwechselnd.
